templates/tools/dockerfile/test/sanity/Dockerfile.template

```diff
@@ -48,9 +48,10 @@
   #======================================
   # More sanity test dependencies (bazel)
   RUN apt-get install -y openjdk-8-jdk
-  # TOOD(jtattermusch): pin the bazel version
+  # Revert Bazel to an old commit that supports running ./compile.sh 
+  # without a local protoc dependency
   RUN git clone https://github.com/bazelbuild/bazel.git /bazel
-  RUN cd /bazel && ./compile.sh
+  RUN cd /bazel && git reset --hard 19ece7c663cd99d338306e302e0f7ce023becfe3 && ./compile.sh
```

Comment:
how about just `git checkout release-0.4.1`? That seems stable enough for our needs.
Also, please add TODO explaining why we need to pin here (because at some point we might need to fix this).